EZSwiftExtensionsTests/DateTests.swift

```diff
@@ -40,17 +40,26 @@ class DateTests: XCTestCase {
         // Given
         let expectedResult = Date(timeIntervalSince1970: 784_111_777)
         let fromGMT = TimeInterval(NSTimeZone.local.secondsFromGMT())
+        let fromStartOfDay = TimeInterval(8 * 3600 + 49 * 60 + 37) // seconds from start of day
 
         // When
         let rfc1123 = Date(httpDateString: "Sun, 06 Nov 1994 08:49:37 GMT")
         let rfc850 = Date(httpDateString: "Sunday, 06-Nov-94 08:49:37 GMT")
         let asctime = Date(httpDateString: "Sun Nov  6 08:49:37 1994")
+        let iso8601DateOnly = Date(httpDateString: "1994-11-06")
```

Comment:
Then I presume even those tests are bad. 

Regardless, these new tests should contain some sort of distinction between the different extensions you are proposing, based on the granularity.